### ISSUE TITLE
move hook for transition-group

### ISSUE THREAD

**Issue by shshaw** [vuejs/vue#6038]
### What problem does this feature solve?
Currently with `<transition-group>` you can build great JS-only transitions, but there's no way to hook into Vue's move functionality. 

A hook/callback ( `move` or `beforeMove` ?) would be extremely handy allowing for custom JS to handle the FLIP when items need to move.

The hook would ideally receive an array of elements that will be moving and an array with the calculated positions/offset that Vue already builds for the CSS FLIP, and if Vue needs it, a `done` callback.

Based on a quick look at the `<transition-group>` code, this should not add much weight since Vue is already collecting all of that data.

Possibly related to #5813 or #4885 but this seems distinct.

### What does the proposed API look like?
```
<transition-group name="flip-me" @move="myFlipMethod"> ... </transition-group>
```

```
  methods: {
 
    // Hook receives an array of elements and an array of the position data
    myFlipMethod(elements, positions, done) {
       // Custom GSAP animation to move the elements, allowing for staggering and fancy effects
       var tl = new TimelineLite({ onComplete: done });
       elements.forEach( (el,i) => {
         tl.from( el, 1, { 
           x: positions[i].dx, 
           y: positions[i].dy, 
           ease: Elastic.easeOut,
           delay: i * 0.01
         });
       });
    }
```

**Comment by Kingwl** [comment]
as @yyx990803 said in https://github.com/vuejs/vue/pull/5813#issuecomment-312144504
so i close this issue

**Comment by yyx990803** [comment]
Unfortunately, FLIP animations are entirely CSS transition driven and it would be very difficult to provide a JS driven API with the same behavior. It would also likely bloat the codebase further. In the future we may split the transition system out as standalone packages so that we can add more opt-in features to it.

**Comment by rowild** [comment]
However, is there any callback at the moment? There are scenarios like recalculating a scrollbar, e.g., where such a final callback would be really handy...

**Comment by mschlitz-trux** [comment]
It would be nice to have something along the lines of "beforeMove" or "afterMove" to apply any kind of pre/post logic desired. For example, it would be nice to be able to set the zIndex of all my elements before and after a transition, because I want a column-stacked list to have the higher cards move "behind" the lower ones when re-organizing.

**Comment by drmercer** [comment]
My use case: a card in a list is moving into a "focused" position, and I want to show some extra details on the card once it stops moving.

A workaround I found, in case this helps anyone else:

* As a substitute for `@before-move`, I just added the relevant logic (`isMoving = true`) to the code that triggers the transition. This feels gross, but it works.
* As a substitute for `@after-move`, I can use the [`transitionend` HTML event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event) that fires on the card element itself. (I can't use `transitionstart` or `transitionrun` for the previous bullet, though, because that's not early enough, so the extra details flash on the card before the transition begins.)
  * In my specific case, since I have other cards in the list, it's easier to just hook into `@after-leave` for this purpose instead, which fires when the other cards leave the list (which happens at the same time as the focused card reaching its position).
